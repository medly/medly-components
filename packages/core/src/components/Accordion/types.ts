export type StaticProps = {
    Header: React.FC;
    Content: React.FC;
    Context: React.Context<AccordionContextType>;
};

export type AccordionContextType = [
    boolean /** Current active state */,
    React.Dispatch<React.SetStateAction<boolean>> /** Function to be called to toggle the active state */
];
