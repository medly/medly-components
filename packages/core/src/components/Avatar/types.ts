import { AvatarSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Size of the avatar */
    size?: AvatarSizes;
    /** Text color */
    textColor?: string;
    /** Background color */
    bgColor?: string;
    /** Hover Text color */
    hoverTextColor?: string;
    /** Hover Background color */
    hoverBgColor?: string;
    /** Hover text shadow color*/
    hoverTextShadowColor?: string;
    /** Hover image shadow color */
    hoverImgShadowColor?: string;
    /** Pressed Background color */
    pressedBgColor?: string;
    /** Pressed Text color */
    pressedTextColor?: string;
}

export interface StyledProps extends AvatarProps {
    /** To check children is image */
    isImage?: boolean;
}
