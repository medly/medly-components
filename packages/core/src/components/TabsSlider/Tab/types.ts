import { TabsSliderTheme } from '@medly-components/theme';
import { HTMLProps, WithThemeProp } from '@medly-components/utils';

export type StyledProps = HTMLProps<HTMLButtonElement> &
    WithThemeProp &
    TabsSliderTheme & {
        active?: boolean;
        fraction?: any;
        totalTabs?: number;
    };

export interface Props extends HTMLProps<HTMLButtonElement> {
    /** Ids of all tabs */
    totalTabs?: number;
    /** Id of tab */
    id: any;
    /** Label of the tab */
    label: string;
    /** Disabled label text */
    disabledLabel?: string;
    /** To be used to hide the tab */
    hide?: boolean;
    /** Disabled State */
    disabled?: boolean;
    /** This will be handled internally */
    active?: boolean;
    /** Provide the fraction of the of the total width of the parent */
    fraction?: number;
}
