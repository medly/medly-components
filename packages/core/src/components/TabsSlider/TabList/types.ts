import { HTMLProps } from '@medly-components/utils';

export type StyledProps = HTMLProps<HTMLButtonElement> & {
    active: number;
    totalTabs?: number;
};

export interface Props extends HTMLProps<HTMLDivElement> {
    active: any;
    onChange: (id: any) => void;
}
