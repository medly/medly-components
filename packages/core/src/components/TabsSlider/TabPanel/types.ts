import { HTMLProps } from '@medly-components/utils';

export interface Props extends HTMLProps<HTMLDivElement> {
    active: any;
}
