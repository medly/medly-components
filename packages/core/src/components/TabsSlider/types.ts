import { HTMLProps, WithStyle } from '@medly-components/utils';
import { Props as TabProps } from './Tab/types';
import { Props as TabListProps } from './TabList/types';
import { Props as TabPanelProps } from './TabPanel/types';

export interface Props extends HTMLProps<HTMLDivElement> {
    /** Id of the default active tab */
    defaultActive?: any;
    /** Id of the active tab */
    active?: any;
    /**  Function to be called on selecting any tab */
    onChange?: (id: any) => void;
}

export interface StaticProps {
    Tab?: React.FC<TabProps> & WithStyle;
    TabList?: React.FC<TabListProps> & WithStyle;
    TabPanel?: React.FC<TabPanelProps> & WithStyle;
}
