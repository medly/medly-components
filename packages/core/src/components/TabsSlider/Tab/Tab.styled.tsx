import { css, styled } from '@medly-components/utils';
import Text from '../../Text';
import { StyledProps } from './types';

export const Label = styled(Text)`
    line-height: normal;
    text-align: center;
`;

export const DisabledLabel = styled(Text)`
    font-size: 1rem;
    line-height: 1.2rem;
    letter-spacing: -0.04rem;
    color: ${({ theme }) => theme.colors.grey[400]};
    font-weight: 400;
    text-align: center;
`;

const getStyle = ({ styleType, labelColor }: StyledProps & { styleType: 'active' | 'default' | 'hovered' | 'disabled' }) => css`
    ${Label} {
        color: ${labelColor[styleType]};
        line-height: ${styleType === 'disabled' && '1.6rem'};
    }
`;

const activeStyle = () => css<StyledProps>`
    ${props => getStyle({ ...props, styleType: 'active' })}
`;

const nonActiveStyle = ({ bgColor }: StyledProps) => css<StyledProps>`
    ${props => getStyle({ ...props, styleType: 'default' })}
    &:not(:disabled):hover {
        ${props => getStyle({ ...props, styleType: 'hovered' })}
        background-color: ${bgColor.hovered};
        cursor: pointer;
    }
`;

const disabledStyle = css<StyledProps>`
    ${props => getStyle({ ...props, styleType: 'disabled' })}
`;

export const TabWrapper = styled('button').attrs(({ theme }) => ({ ...theme.tabsSlider }))<StyledProps>`
    width: calc(${({ totalTabs }) => 100 / totalTabs}% - ${({ totalTabs }) => (totalTabs * 0.4) / (totalTabs - 1)}rem);
    user-select: none;
    text-decoration: none;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-flow: row wrap;
    position: relative;
    border: none;
    box-sizing: border-box;
    font-family: inherit;
    border-radius: 0.8rem;
    text-align: center;
    padding: 0.8rem 0;
    flex: ${({ fraction }) => fraction};
    background: transparent;
    transition: all 100ms ease-out;
    min-height: 100%;
    * {
        transition: all 100ms ease-out;
    }

    &:focus {
        outline: none;
    }

    &:disabled {
        cursor: not-allowed;
        padding: 0 4rem;
        &::after {
            background-color: transparent;
        }
        ${disabledStyle}
    }

    ${({ active }) => (active ? activeStyle : nonActiveStyle)}
`;

export const LabelAndDetailsWrapper = styled.div`
    display: flex;
    flex-direction: column;
`;

export const LabelWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;
