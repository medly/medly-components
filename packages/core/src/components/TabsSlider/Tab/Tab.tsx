import { WithStyle } from '@medly-components/utils';
import React from 'react';
import * as Styled from './Tab.styled';
import { Props } from './types';

const disabledLabel = 'Not available';

export const Tab: React.FC<Props> & WithStyle = React.memo(props => {
    const { id, active, disabled, label, ...restProps } = props;

    return (
        <Styled.TabWrapper
            {...{ id, active, disabled, ...restProps }}
            type="button"
            role="tab"
            aria-selected={active ? 'true' : 'false'}
            aria-controls={`panel-${id}`}
            tabIndex={active ? 0 : -1}
        >
            <Styled.LabelAndDetailsWrapper>
                <Styled.LabelWrapper>
                    <Styled.Label id={`${id}-label`} textWeight="Medium" textVariant="body2">
                        {label}
                    </Styled.Label>
                    {disabled && (
                        <Styled.DisabledLabel textWeight="Medium" textVariant="body2" id={`${id}-disabledLabel`}>
                            {disabledLabel}
                        </Styled.DisabledLabel>
                    )}
                </Styled.LabelWrapper>
            </Styled.LabelAndDetailsWrapper>
        </Styled.TabWrapper>
    );
});
Tab.displayName = 'Tab';
Tab.Style = Styled.TabWrapper;
