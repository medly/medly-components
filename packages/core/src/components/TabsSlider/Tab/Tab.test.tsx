import { render } from '@test-utils';
import React from 'react';
import { Tab } from './Tab';
import { Props } from './types';

const renderer = ({ id = 'dummy', label = 'Add', hide = false, disabled = false, active = false }: Partial<Props>) =>
    render(<Tab {...{ id, label, hide, disabled, active }}>Content for the add panel</Tab>);

describe('Tab', () => {
    it('should render active state properly', () => {
        const { container } = renderer({ active: true });
        expect(container).toMatchSnapshot();
    });

    it('should render disabled state properly', () => {
        const { container } = renderer({ disabled: true });
        expect(container).toMatchSnapshot();
    });
});
