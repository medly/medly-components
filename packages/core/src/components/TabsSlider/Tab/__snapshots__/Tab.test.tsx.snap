// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render active state properly 1`] = `
.c3 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c5 {
  line-height: normal;
  text-align: center;
}

.c0 {
  width: calc(NaN% - NaNrem);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  position: relative;
  border: none;
  box-sizing: border-box;
  font-family: inherit;
  border-radius: 0.8rem;
  text-align: center;
  padding: 0.8rem 0;
  background: transparent;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  min-height: 100%;
}

.c0 * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  padding: 0 4rem;
}

.c0:disabled::after {
  background-color: transparent;
}

.c0:disabled .c4 {
  color: #b0bcc8;
  line-height: 1.6rem;
}

.c0 .c4 {
  color: #126AFA;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-controls="panel-dummy"
    aria-selected="true"
    class="c0"
    id="dummy"
    role="tab"
    tabindex="0"
    type="button"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3 c4 c5"
          id="dummy-label"
        >
          Add
        </span>
      </div>
    </div>
  </button>
</div>
`;

exports[`Tab should render disabled state properly 1`] = `
.c3 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c5 {
  line-height: normal;
  text-align: center;
}

.c6 {
  font-size: 1rem;
  line-height: 1.2rem;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  color: #b0bcc8;
  font-weight: 400;
  text-align: center;
}

.c0 {
  width: calc(NaN% - NaNrem);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  position: relative;
  border: none;
  box-sizing: border-box;
  font-family: inherit;
  border-radius: 0.8rem;
  text-align: center;
  padding: 0.8rem 0;
  background: transparent;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  min-height: 100%;
}

.c0 * {
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: not-allowed;
  padding: 0 4rem;
}

.c0:disabled::after {
  background-color: transparent;
}

.c0:disabled .c4 {
  color: #b0bcc8;
  line-height: 1.6rem;
}

.c0 .c4 {
  color: #607890;
}

.c0:not(:disabled):hover {
  background-color: #dfe4e9;
  cursor: pointer;
}

.c0:not(:disabled):hover .c4 {
  color: #546A7F;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <button
    aria-controls="panel-dummy"
    aria-selected="false"
    class="c0"
    disabled=""
    id="dummy"
    role="tab"
    tabindex="-1"
    type="button"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3 c4 c5"
          id="dummy-label"
        >
          Add
        </span>
        <span
          class="c3 c6"
          id="dummy-disabledLabel"
        >
          Not available
        </span>
      </div>
    </div>
  </button>
</div>
`;
